libp2p/stream/bufferstream.nim

```diff
@@ -226,12 +254,14 @@ method write*(s: BufferStream, msg: seq[byte]): Future[void] =
   ## If ``msglen > len(sbytes)`` only ``len(sbytes)`` bytes will be written to
   ## stream.
   ##
+
+  if s.closed:
+    raise newLPStreamClosedError()
+
   if isNil(s.writeHandler):
-    var retFuture = newFuture[void]("BufferStream.write(seq)")
-    retFuture.fail(newNotWritableError())
-    return retFuture
+    raise newNotWritableError()
```

Comment:
is there any buffer stream that is not writeable?

Reply:
yeah, if no `writeHandler` then it will throw, it isn't handled anywhere specifically, just by the `CatchableError` exceptions.